src/Pages/Main/Components/MainCarousel/MainCarousel.js

```diff
@@ -0,0 +1,264 @@
+import { Link, withRouter } from 'react-router-dom';
+import React, { useState, useEffect, Fragment } from 'react';
+import styled, { keyframes } from 'styled-components';
+import { ArrowIosBackOutline } from '@styled-icons/evaicons-outline';
+import { ArrowIosForwardOutline } from '@styled-icons/evaicons-outline';
+
+function MainCarousel() {
+  const [mainCarouselDatas, setMainCarouselDatas] = useState();
+  const [index, setIndex] = useState(0);
+  const delay = 4000;
+
+  useEffect(() => {
+    fetch('data/mainCarouselData.json')
+      .then(res => res.json())
+      .then(carouselDatas => setMainCarouselDatas(carouselDatas.courses));
+  }, []);
+
+  useEffect(() => {
+    mainCarouselDatas &&
+      setTimeout(
+        () =>
+          setIndex(prevIndex =>
+            prevIndex === mainCarouselDatas.length - 1 ? 0 : prevIndex + 1
+          ),
+        delay
+      );
+    // return () => {};
+  }, [mainCarouselDatas, index]);
```

Comment:
```suggestion
const startSlide = () => {
    mainCarouselDatas &&
      setIndex(prevIndex =>
        prevIndex === mainCarouselDatas.length - 1 ? 0 : prevIndex + 1
      );
  };

  useEffect(() => {
    setInterval(startSlide, delay);
    return () => clearInterval(startSlide);
  }, [mainCarouselDatas]);
```

1. timeout을 interval로 변경했습니다!
2. 어제 말씀하신 부분중 return 하는 부분 말씀하셨는데 이 부분이였나 보네요!
3. 해당 부분은 componentWillUnmount 로서 인터벌을 종료하는 로직입니다.

Reply:
4. 의존성 배열에 변화가 생겼죠?

Reply:
해당 useEffect가 어느 시점에 실행되는지 비교해보시기 바랍니다!!

Reply:
감사한니다 도현님 !setInterval로 바꿀생각을 못했었네요 ㅠㅠ